src/timer.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * A timer can be set for a number or an animation.
+ * Numbers will be decremented by the game clock tick every update.
+ * Animations will execute upon the final frame.
+ *
+ * Once the timer executes, it will call the toggleFunction.
+ * toggleThis will be the value of this provided for the call to toggleFunction.
+ * toggleFunctionArguments is an array of arguments that will be passed in (but not as an array).
+ */
+
+function Timer(game, source, startOnCreate, toggleFunction, toggleThis, toggleFunctionArguments) {
+    if (typeof source !== "number" && source.isDone() === undefined) {
+        console.log("Timer Error: Timer source must be either a number or an animation.");
+        return;
+    }
+    
+    if (typeof toggleFunction !== "function") {
+        console.log("Timer Error: toggleFunction is not a function.");
+        return;
+    }
+    
+    this.game = game;
+    this.source = source;
+    this.toggleFunction = toggleFunction;
+    this.toggleThis = toggleThis;
+    this.toggleFunctionArguments = toggleFunctionArguments;
+    
+    if (startOnCreate) {
```

Comment:
Assuming startOnCreate is always boolean, this can be simplified as:
    this.isDone = startOnCreate;

    if (typeof this.source === "number") {
        this.initialValue = this.source;
    } else {
        this.initialValue = 0;
    }